app/cms/github-webhook.server.ts

```diff
@@ -32,6 +34,70 @@ export function pushEventCacheKeysToInvalidate(
     }
   )
 
+  return matchingDocCollections
+}
+
+function getManifestPathForDocCollection(docCollection: any): string {
+  return posix.join(docCollection.source.rootPath, JSON_MANIFEST_FILENAME)
+}
+
+function getChangedFilesFromCommits(commits: Commit[]): string[] {
+  return commits.flatMap((commit) => [
+    ...commit.added,
+    ...commit.removed,
+    ...commit.modified,
+  ])
+}
+
+function getPathsOfDocumentsFromChanges(
+  docCollection: any,
+  allChangedFiles: string[]
+): string[] {
+  return allChangedFiles.filter((path) => {
+    let isDocumentPath = path.startsWith(docCollection.source.rootPath)
+    let isDocument = path.endsWith(".md") || path.endsWith(".mdx")
```

Comment:
There's a `filterFileNameHasMarkdownExtension` at `app/cms/github/filter-file-name-has-markdown-extension.ts` - might be better to use that here? (Though looking at it, the regex should probably be modified with the "i" option, i.e. ` /(.mdx?)/i`, to ignore case).

Reply:
This code and the above example were already written in this file, I just moved them into functions.